### ISSUE TITLE
linter: `no-unused-vars`: `args: "after-used"` incorrectly reports parameters before a used rest parameter

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26562]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "rules": {
    "typescript/no-unused-vars": ["error", { "args": "after-used" }]
  }
}
```

### What happened?

```ts
function consumeRest(unusedBeforeRest: string, ...usedRest: Array<string>) {
  console.log(usedRest);
}

consumeRest('unused', 'used');
```

## Actual result

```text
eslint(no-unused-vars): Parameter 'unusedBeforeRest' is declared but never used.
```

## Expected result

No error, matching ESLint and `@typescript-eslint/no-unused-vars`.

With `args: "after-used"`, unused parameters before the last used parameter are allowed. Here,
`usedRest` is the last parameter and is used, so `unusedBeforeRest` should not be reported.

Oxlint accepts `after-used`, and its configuration schema lists it as a supported `args` value.
This therefore appears to be an implementation bug rather than an unsupported option.

**Comment by lukpsaxo** [comment]
similiar but different error previously fixed: https://github.com/oxc-project/oxc/issues/22625
